### PR TITLE
feat: add pr automation controller

### PR DESCRIPTION
**Pull request by floreks**
## Summary

- Bump `kubebuilder` to `v.0.14.0`. This cause all CRD schema to slightly change
- Refactor `Status` into a common struct as most of our resources uses the same status object and methods
- Simplify reading our CRD references and their ID
- Add a new `PrAutomation` controller
- Add a sample `pr_automation.yaml`
- Improve controller tests (currently only PrAutomation uses it) to be able to run tests in parallel instead of in order

## Test Plan

Running controller locally on existing `https://console.cd-demo.onplural.sh` API.

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

**Review by michaeljguarino — COMMENTED** [review]
I think we need to also add the `creates` field which was fleshed out recently (in addition to `updates`)

File: controller/api/v1alpha1/prautomation_types.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+func (in *PrAutomation) Attributes(clusterID *string, serviceID *string, connectionID *string) *console.PrAutomationAttributes {
+	attrs := console.PrAutomationAttributes{
+		Name:          lo.ToPtr(in.ConsoleName()),
+		Identifier:    in.Spec.Identifier,
+		Documentation: in.Spec.Documentation,
+		Title:         in.Spec.Title,
+		Message:       in.Spec.Message,
+		Branch:        in.Spec.Branch,
+		Updates:       in.Spec.Updates.Attributes(),
```
Comment (michaeljguarino) [inline controller/api/v1alpha1/prautomation_types.go:76]: in particular here doesn't pass the creates spec

Comment (floreks) [reply controller/api/v1alpha1/prautomation_types.go:76]: oh, I must have missed it

**Comment by floreks** [comment]
> I think we need to also add the `creates` field which was fleshed out recently (in addition to `updates`)

Attributes were also missing repository ID. That should cover it https://github.com/pluralsh/console/pull/656/commits/e8b4a6a0afdab9f34acd24bfc843bd388ba40bfa.

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**